src/ds3.c

```diff
@@ -1016,6 +1021,12 @@ void ds3_request_set_custom_header(ds3_request* _request, const char* header_nam
 }
 
 void ds3_request_set_metadata(ds3_request* _request, const char* name, const char* value) {
+    if ((value == NULL)
+      ||(strlen(value) == 0)) {
+        fprintf(stderr, "Ignoring NULL or empty entry for key %s\n", name);
```

Comment:
Changed to:        
fprintf(stderr, "Ignoring metadata key \"%s\" which has a NULL or empty value.\n", name);